### PR TITLE
[JENKINS-49075] Bump parent pom and a few other dependencies

### PR DESCRIPTION
**Pull request by rysteboe**
@reviewbybees 

@jglick, bumping the required core version just to satisfy the PCT doesn't sit well with me, but filing the PR anyway for you to accept or reject.

https://issues.jenkins-ci.org/browse/JENKINS-49075

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java [hunk 1]
```diff
@@ -79,8 +80,9 @@ private DockerFingerprints() {} // no instantiation
      * @return Created fingerprint or null if it is not found
      * @throws IOException Fingerprint loading error
      */
+    @SuppressFBWarnings("RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE")
     public static @CheckForNull Fingerprint of(@Nonnull String id) throws IOException {
-        final Jenkins jenkins = Jenkins.getInstance(); // should be not null
+        final Jenkins jenkins = Jenkins.getInstance(); // should be not null, but technically can be
```
Comment (oleg-nenashev) [inline src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java]: Use `getInstanceOrNull` then

File: pom.xml [hunk 2]
```diff
@@ -31,9 +31,8 @@
   </scm>
 
   <properties>
-    <jenkins.version>1.642.4</jenkins.version>
-    <java.level>7</java.level>
-    <jenkins-test-harness.version>2.30</jenkins-test-harness.version>
+    <jenkins.version>2.60.3</jenkins.version>
```
Comment (jglick) [inline pom.xml]: Any particular reason you are pushing it this high? I have no problem with it, but you seemed to be saying that you felt compelled to do so, which I am not following.

File: src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java (hunk 1 above)
Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java]: Right, or drop the null check—I see no reason for this code to be run after Jenkins shutdown so the null check should not be necessary.

File: pom.xml (hunk 2 above)
Comment (rysteboe) [reply pom.xml]: The PCT fails without this bump.  I can do more investigation into alternative fixes, but this was the only way that I could fix it without spending an undue amount of time working on this.

File: src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java [hunk 3]
```diff
@@ -79,9 +80,8 @@ private DockerFingerprints() {} // no instantiation
      * @return Created fingerprint or null if it is not found
      * @throws IOException Fingerprint loading error
      */
-    public static @CheckForNull Fingerprint of(@Nonnull String id) throws IOException {
-        final Jenkins jenkins = Jenkins.getInstance(); // should be not null
-        return jenkins != null ? jenkins.getFingerprintMap().get(getFingerprintHash(id)) : null;
+    public static Fingerprint of(@Nonnull String id) throws IOException {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java]: You need to leave `@CheckForNull` on this method—there may be no such fingerprint. (I think you got confused by removing the code path where `Jenkins.instance == null`.)

Comment (rysteboe) [reply src/main/java/org/jenkinsci/plugins/docker/commons/fingerprint/DockerFingerprints.java]: Yeah fixed.  Didn't look closely enough at the rest of the code.

**Pull request merged**